src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -248,31 +232,32 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @param {Object} file - Box file
      * @return {string} Sidebar view to use
      */
-    getDefaultSidebarView(props: Props, file?: BoxItem, editors?: Array<MetadataEditor>): SidebarView {
-        const { view, hasBeenToggled }: State = this.state;
-        const { isLarge, defaultView }: Props = props;
+    getDefaultSidebarView(file?: BoxItem, metadataEditors?: Array<MetadataEditor>): SidebarView {
```

Comment:
We had talked previously about moving this to being derived in `render` rather than living in state. It seems like that would clean up the logic around current/default view considerably. Do you want to tackle that as part of this change?

Reply:
I mentioned to you that we cant do that. We need to keep view in state, since the future view depends upon the prior view. Also, once new view is calculated, that needs to be set to the state, hence it cannot be part of render